src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -1100,142 +1107,98 @@ static int create_vni_context(int vni)
-
-            av_attr.name = NULL;
-            av_attr.flags = 0;
-            MPIDI_OFI_CALL(fi_av_open(domain, &av_attr, &av, NULL), avopen);
-        }
-
-        MPIDI_OFI_global.domain = domain;
-        MPIDI_OFI_global.av = av;
-        MPIDI_OFI_global.rma_cmpl_cntr = rma_cmpl_cntr;
-    } else {
-        /* for vni > 0, shared objects already created */
-        domain = MPIDI_OFI_global.domain;
-        av = MPIDI_OFI_global.av;
-        rma_cmpl_cntr = MPIDI_OFI_global.rma_cmpl_cntr;
-    }
-
     struct fid_cq *cq;
 
-    struct fi_cq_attr cq_attr;
-    memset(&cq_attr, 0, sizeof(cq_attr));
-    cq_attr.format = FI_CQ_FORMAT_TAGGED;
-    MPIDI_OFI_CALL(fi_cq_open(domain, &cq_attr, &cq, NULL), opencq);
-
     struct fid_ep *ep;
     struct fid_ep *tx;
     struct fid_ep *rx;
 
-    if (MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS) {
-        if (vni == 0) {
-            /* shared components */
-            MPIDI_OFI_CALL(fi_scalable_ep(domain, prov_use, &ep, NULL), ep);
-            MPIDI_OFI_CALL(fi_scalable_ep_bind(ep, &av->fid, 0), bind);
-        }
-
-        struct fi_tx_attr tx_attr;
-        tx_attr = *prov_use->tx_attr;
-        tx_attr.op_flags = FI_COMPLETION;
-        if (MPIDI_OFI_ENABLE_RMA || MPIDI_OFI_ENABLE_ATOMICS)
-            tx_attr.op_flags |= FI_DELIVERY_COMPLETE;
-        tx_attr.caps = 0;
-
-        if (MPIDI_OFI_ENABLE_TAGGED)
-            tx_attr.caps = FI_TAGGED;
-
-        /* RMA */
-        if (MPIDI_OFI_ENABLE_RMA)
-            tx_attr.caps |= FI_RMA;
-        if (MPIDI_OFI_ENABLE_ATOMICS)
-            tx_attr.caps |= FI_ATOMICS;
-        /* MSG */
-        tx_attr.caps |= FI_MSG;
-        tx_attr.caps |= FI_NAMED_RX_CTX;        /* Required for scalable endpoints indexing */
-
-        MPIDI_OFI_CALL(fi_tx_context(ep, vni, &tx_attr, &tx, NULL), ep);
-        MPIDI_OFI_CALL(fi_ep_bind(tx, &cq->fid, FI_SEND | FI_SELECTIVE_COMPLETION), bind);
-        MPIDI_OFI_CALL(fi_ep_bind(tx, &rma_cmpl_cntr->fid, FI_WRITE | FI_READ), bind);
-
-        struct fi_rx_attr rx_attr;
-        rx_attr = *prov_use->rx_attr;
-        rx_attr.caps = 0;
-
-        if (MPIDI_OFI_ENABLE_TAGGED) {
-            rx_attr.caps |= FI_TAGGED;
-            rx_attr.caps |= FI_DIRECTED_RECV;
-        }
-
-        if (MPIDI_OFI_ENABLE_RMA)
-            rx_attr.caps |= FI_RMA | FI_REMOTE_READ | FI_REMOTE_WRITE;
-        if (MPIDI_OFI_ENABLE_ATOMICS)
-            rx_attr.caps |= FI_ATOMICS;
-        rx_attr.caps |= FI_MSG;
-        rx_attr.caps |= FI_MULTI_RECV;
-        rx_attr.caps |= FI_NAMED_RX_CTX;        /* Required for scalable endpoints indexing */
-
-        MPIDI_OFI_CALL(fi_rx_context(ep, vni, &rx_attr, &rx, NULL), ep);
-        MPIDI_OFI_CALL(fi_ep_bind(rx, &cq->fid, FI_RECV), bind);
+#ifdef MPIDI_OFI_VCI_USE_DOMAIN
+    mpi_errno = create_vni_domain(&domain, &av, &rma_cmpl_cntr);
+    MPIR_ERR_CHECK(mpi_errno);
+    mpi_errno = create_cq(domain, &cq);
+    MPIR_ERR_CHECK(mpi_errno);
 
+    if (MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS) {
```

Comment:
@hzhou I think some high level code comments would be helpful. (here, and in other places with big code changes and missing comments)

Reply:
You are right. I'll add another commit to refactor the comments.